src/content/1.7/modules/concepts/controllers/admin-controllers/tabs.md

```diff
@@ -227,9 +238,11 @@ class example_module_mailtheme extends Module
         $tab = new Tab($tabId);
         $tab->active = 1;
         $tab->class_name = 'MyModuleDemoController';
+        // Only since 1.7.7, you can define a route name
+        $tab->route_name = 'admin_my_symfony_routing';
         $tab->name = array();
         foreach (Language::getLanguages() as $lang) {
-            $tab->name[$lang['id_lang']] = 'My Module Demo';
+            $tab->name[$lang['locale']] = $this->trans('My Module Demo', array(), 'Modules.MyModule.Admin', $lang['locale']);
```

Comment:
I'm pretty sure multilingual fields are indexed using `id_lang`
```suggestion
            $tab->name[$lang['id_lang']] = $this->trans('My Module Demo', array(), 'Modules.MyModule.Admin', $lang['locale']);
```

Reply:
Yep so was I, which is why it was documented this way But actually when the `ModuleTabRegister` manages this property it automatically converts it with the appropriate id, it is able to do it with `locale` `language_code` and `iso_code`

So `id_lang` is the only one that doesn't work.. It will only get the first one in the array so all tabs have the same name

Reply:
https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Module/Tab/ModuleTabRegister.php#L251-L271

Reply:
It seems you're not talking about the same thing.

In this code we deal with the ObjectModel `Tab`. Like all the other objects, it requires the id_lang to match a string for a specific lang.

This link (https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Module/Tab/ModuleTabRegister.php#L251-L271) is for automatic registration. :)

Reply:
Ooohh you're right I mixed it up The locale must be used only in `$tabs` I'm gonna fix this

Reply:
Done